### PR TITLE
modify getContainer and delContainer methods in BfCoordWorkHelper to use WorkMgrServiceV2

### PR DESCRIPTION
**Pull request by kidsbear**
Also introduces a subResource class - `ContainerResource`, only add a limit amount api calls in `ContainerResource` that can be supported now. Next step: introduce a new structure to support container creation, and provide api call `createContainer` in `WorkMgrServiceV2`.

### CODE REVIEW

File: projects/coordinator/src/main/java/org/batfish/coordinator/WorkMgrServiceV2.java [hunk 1]
```diff
@@ -28,40 +27,39 @@
...
 
   /** Returns the list of {@link Container containers} that the given API key may access. */
   @GET
-  @Path("/containers")
-  public Response getContainers(
-      @DefaultValue(DEFAULT_API_KEY) @HeaderParam(SVC_KEY_API_KEY) String apiKey) {
-    List<Container> containers = Main.getWorkMgr().getContainers(apiKey);
+  @Path(CoordConsts.SVC_KEY_CONTAINERS)
+  public Response getContainers() {
+    _logger.info("WMS:getContainers\n");
```
Comment (dhalperi) [inline projects/coordinator/src/main/java/org/batfish/coordinator/WorkMgrServiceV2.java]: WMS2

File: projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import javax.ws.rs.core.UriInfo;
+import org.batfish.common.BatfishLogger;
+import org.batfish.common.Container;
+import org.batfish.coordinator.Main;
+
+/**
+ * The {@link ContainerResource} is a resource for servicing client API calls at container level.
+ *
+ * <p>The ContainerResource provides information about a specified container, and provides the
+ * ability to create/delete a specified container for authenticated clients.
```
Comment (dhalperi) [inline projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java]: does not have ability to create a container -- that goes somewhere else. (in the ContainersResource)

File: projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    validate();
+    if (Main.getWorkMgr().delContainer(_name)) {
+      return Response.noContent().build();
+    } else {
+      return Response.serverError().build();
+    }
+  }
+
+  /** Validates the container {@link #_name} exists and {@link #_apiKey} has accessibility to it. */
+  private void validate() {
```
Comment (dhalperi) [inline projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java]: checkAccessToContainer

File: projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    java.nio.file.Path containerDir = Main.getSettings().getContainersLocation().resolve(_name);
```
Comment (dhalperi) [inline projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java]: is there a WorkManager function you can use for this instead?

One proposal would be to use a function like

```java
@Nullable getContainer(String name);
```

then you throw the right exception type if it's null.

But that's just a random thought, might not work well in practice, think about it.

File: projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    if (!Files.exists(containerDir)) {
+      throw new NotFoundException(String.format("Container '%s' does not exist", _name));
```
Comment (dhalperi) [inline projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java]: another thing you could do is just

```java
List<Container> listContainers(String apiKey)
```

and check if the list contains the specified container.

It's okay to not return a Forbidden exception if the user doesn't have access to even list it. Security-wise, it's okay to not leak that a container exists with that name.

File: projects/coordinator/src/test/java/org/batfish/coordinator/WorkMgrServiceV2Test.java [hunk 6]
```diff
@@ -54,32 +60,48 @@ protected Application configure() {
   }
 
   @Test
-  public void getContainers() throws Exception {
-    Response response = target("/v2/containers").request().get();
+  public void getContainers() {
+    Response response = target(BASE_CONTAINER_PATH).request().get();
```
Comment (dhalperi) [inline projects/coordinator/src/test/java/org/batfish/coordinator/WorkMgrServiceV2Test.java]: target(V2_CONSTANT).path(CONTAINERS)
target(V2_CONSTANT).path(CONTAINER)
target(V2_CONSTANT).path(CONTAINER).path(ID)

(something like that)

and then you don't need to use Paths to join URLs -- that's what chained calls to target are for.

Comment (dhalperi) [reply projects/coordinator/src/test/java/org/batfish/coordinator/WorkMgrServiceV2Test.java]: or you could make a private static function getContainers() that returns a Response and contains this whole line (with proper targeting)

and another one getContainer(String id)
...

File: projects/batfish-client/src/main/java/org/batfish/client/BfCoordWorkHelper.java [hunk 7]
```diff
@@ -152,22 +157,24 @@ public boolean delAnalysis(String containerName, String analysisName) {
 
   public boolean delContainer(String containerName) {
     try {
-      WebTarget webTarget = getTarget(CoordConsts.SVC_RSC_DEL_CONTAINER);
+      WebTarget webTarget =
+          getTargetV2(Lists.newArrayList(CoordConsts.SVC_KEY_CONTAINERS, containerName));
```
Comment (dhalperi) [inline projects/batfish-client/src/main/java/org/batfish/client/BfCoordWorkHelper.java]: this should be `container/name`, not `containers/name`.

Comment (kidsbear) [reply projects/batfish-client/src/main/java/org/batfish/client/BfCoordWorkHelper.java]: I think Github is using plural form for all resource nouns. [get-single-issue](https://developer.github.com/v3/issues/#get-a-single-issue), [get-single-PR](https://developer.github.com/v3/pulls/#get-a-single-pull-request), [delete-a-comment](https://developer.github.com/v3/issues/comments/#delete-a-comment)

**Pull request merged**